dissect/target/plugins/os/unix/bsd/osx/network.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import plistlib
+from functools import cache, lru_cache
+from typing import Iterator
+
+from dissect.target.helpers.record import MacInterfaceRecord
+from dissect.target.plugins.general.network import NetworkPlugin
+from dissect.target.target import Target
+
+
+class MacNetworkPlugin(NetworkPlugin):
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self._plistlease = cache(self._plistlease)
+        self._plistnetwork = lru_cache(32)(self._plistnetwork)
```

Comment:
I think the 2 cahce types should be switched around as `_plistnetwork()` is the one without extra arguments.